### DISCUSSION TITLE
Obsidian Icon Folder is going to be Iconize!

### DISCUSSION THREAD

**Discussion by FlorianWoelki** [FlorianWoelki/obsidian-icon-folder#232] [Announcements]
We're immensely proud of the progress and accomplishments we've made with the Obsidian icon folder plugin. Each milestone and feature we've rolled out has brought unique value to our users. However, our vision for icons within Obsidian is far more expansive. Why should icons be confined to just folders or files? These subtle visuals serve not just as aesthetic enhancements, but as intuitive guides, making navigation and understanding a seamless process.

Understanding the potential and significance of this, we've embarked on a journey to elevate the user experience even further. Today, we're thrilled to introduce you to "Iconize". This isn't just another plugin or a renaming; it's our commitment to transform and enrich the entire Obsidian ecosystem.

Iconize is our latest venture, designed meticulously to provide a more visually engaging and user-friendly environment in Obsidian. With Iconize, every note, every link, every tab will come to life in an unprecedented way. We're taking the Obsidian experience from monochrome to technicolor, ensuring that your workflow is not just efficient, but also delightful.

Join us on this exciting journey as we reimagine the world of Obsidian with Iconize. Let's make every interaction not just functional, but beautifully iconic!

Two more things: If you already have an idea on what to implement, feel free to create a [discussion in the ideas category](https://github.com/FlorianWoelki/obsidian-iconize/discussions/categories/ideas). This way, users can vote on the idea and we see what users want.

In addition, if you want to support the development and the vision, we love to see your support by [buying us some coffee](https://www.buymeacoffee.com/florianwoelki) ☕❤️

**Comment by BaughBerry** [comment]
I'm confused what this will actually change. Does this mean icons in-line? How will this be distinct from Supercharged Links?

**Reply by FlorianWoelki** [reply]
Well, the main vision is that you can apply Icons everywhere you want. E.g., in the newest properties from Obsidian, the title, or even in the text. We will even enhance the overall functionality and API so that it can be used by other plugins as well.

**Reply by roman-balzer** [reply]
I really love the vision. If it works great it could make **Supercharged Links** and **Icon Shortcodes** obsolete for me. I'd love to know more about what's planned, is there an open discussion to that topic (beside this one) somewhere?

**Reply by FlorianWoelki** [reply]
At the moment, there is nothing really planned out, it's for now a longterm vision and it will probably start with integrating icons in text.

However, if you already have ideas, feel free to post them [here](https://github.com/FlorianWoelki/obsidian-iconize/discussions/categories/ideas).

**Reply by roman-balzer** [reply]
Oh, I thought that maybe there is already a broad outline what these changes could be. Well, I'll throw a list in here, which thoughts I had about Icons. (I don't wanna create a seperate post for each right now).

- Set Callout Icon (+ Color) -> _Replacement for [Callout Manager](https://github.com/eth-p/obsidian-callout-manager)_
- Auto-Icon for external links (GitHub, YT, X, ... - configurable)
- Replace nodes with icons in GraphView (based on a rule-set like [Supercharged Links](https://github.com/mdelobelle/obsidian_supercharged_links))
- Icons in text (_Replacement for [Icon Shortcodes](https://github.com/aidenlx/obsidian-icon-shortcodes)_)
- Icons for properties
- Icons for links (_Replacement for [Supercharged Icons](https://github.com/mdelobelle/obsidian_supercharged_links)_)
  - #38   
- Icon in title (manual, or automatic based on _Icon for Links_) 
  - #190 
- Icons for tags
- Icons for lists (_Replacement for [List Callouts](https://github.com/mgmeyers/obsidian-list-callouts)_)

**Reply by FlorianWoelki** [reply]
Thanks for the list, I'll keep it in mind :)

**Reply by milotek** [reply]
when does it come out?

**Reply by FlorianWoelki** [reply]
There is no ETA at the moment.

**Reply by huyz** [reply]
It'll be interesting to see how this would work with Supercharged Links and Style Settings, which already automatically add icons based on specified criteria

**Reply by ChristinWhite** [reply]
Integrating with or replacing the functionality of Link Favicons would also rock. Although maybe that's already covered by suggestions for Supercharged Links, I'm not very familiar with it.

**Comment by FlorianWoelki** [comment]
**Update**: Icons and Emojis above note titles are now available for version >=2.5.0

**Comment by Thomas-Schmall** [comment]
I have a bit trouble since the update ... one of my vaults gives me a hundred error messages about missing icon packs. I installed and uninstalled the plugin, but it didn't help (at least on mobile). I don't understand where they are supposed to be. Do I really need them? I don't use any.

Also, nearly all icons got deleted via Sync. At some point it left only the ones I used, but not all of those. I had backups and now changed the folder location to the new suggested location, and it seems alright now. But I'm scared it'll happen again.

And then there is the auto-renaming of the icon files. It's removing all spaces (also dashes and underscores? It seems a bit random - is it old behavior or does it still do that?  Is it really necessary? I'm just ending up with double files for no apparent reason.

I like the new features though (like the tab icons) and the plans. Seems good to expand the scope.

**Comment by FlorianWoelki** [comment]
> I have a bit trouble since the update ... one of my vaults gives me a hundred error messages about missing icon packs. I installed and uninstalled the plugin, but it didn't help (at least on mobile). I don't understand where they are supposed to be. Do I really need them? I don't use any.
> 
> Also, nearly all icons got deleted via Sync. At some point it left only the ones I used, but not all of those. I had backups and now changed the folder location to the new suggested location, and it seems alright now. But I'm scared it'll happen again.
> 
> And then there is the auto-renaming of the icon files. It's removing all spaces (also dashes and underscores? It seems a bit random - is it old behavior or does it still do that? Is it really necessary? I'm just ending up with double files for no apparent reason.
> 
> I like the new features though (like the tab icons) and the plans. Seems good to expand the scope.

There is a new background checker that will re-add all the icons for you if some of them are missing. I highly recommend trying that.
There were some issues with the file names which included e.g., spaces. So yes this behavior is intended.
